### DISCUSSION TITLE
Cross-compiling a c++ program containing Magick++.h

### DISCUSSION THREAD

**Discussion by alasdairiainclark** [ImageMagick/ImageMagick#3899] [Help]
I'm trying to create a program which uses the Magick++ library, but I seem to be going around in circles. I'm using Visual Studio Code on Ubuntu 20.04 on my PC, where everything works great. If I try to crosscompile it to arm-linux-gnueabihf to run on my Raspberry Pi I get:
```
[main] Building folder: testProject 
[build] Starting build
[proc] Executing command: /usr/bin/cmake --build /home/user/dev/testProject/build --config Release --target all -j 4 --
[build] [1/1 100% :: 0.040] Linking CXX executable testProject
[build] FAILED: testProject 
[build] : && /usr/bin/arm-linux-gnueabihf-g++-9  -pthread -O3 -DNDEBUG  -rdynamic CMakeFiles/testProject.dir/main.cpp.o  -o testProject  -L/usr/local/lib -lMagick++-7.Q16HDRI -lMagickWand-7.Q16HDRI -lMagickCore-7.Q16HDRI && :
[build] /usr/local/lib/libMagick++-7.Q16HDRI.so: file not recognized: file format not recognized
[build] collect2: error: ld returned 1 exit status
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1
```
As I say it compiles fine for the PC, on the PC. I can also cross-compile a simple "hello world" project without Magick++ and there are no issues. I believe it is because Magick++ isn't set up for cross-compiling, but I don't know how to do this. It could also be an issue with my CMakeLists.txt file, but as it works on the PC I think not.
I hope I've put this in the right place - I've spent about a fortnight trying different things and getting nowhere.

**Comment by urban-warrior** [comment]
Perhaps an ImageMagick user can be helpful.  The development team does provide help with build problems due to complex compiler chains.   There are just too many compilers, IDE's, and host types for us to be knowledgeable in all.
